app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -147,6 +149,9 @@ class HomeFragment : Fragment() {
 
         lifecycle.addObserver(sessionObserver)
 
+        if (finishOnboarding)
```

Comment:
We could make this A) more close to what users experience and B) more robust by adding the override directly to the `FenixOnboarding` class in the function used by the application code, e.g.:

```kotlin
fun userHasBeenOnboarded() = isOverriddenThroughIntent ||
        onboardingPrefs.onboardedVersion == CURRENT_ONBOARDING_VERSION
```

It's more robust such that when someone refactors this code, it's more likely that they won't accidentally break our code.